### ISSUE TITLE
[Linux][Arm] assert currentProcessCpuCount == g_processAffinitySet.Count() in GC/API/GC/KeepAliveNull/KeepAliveNull.sh

### ISSUE THREAD

**Issue by jashook** [dotnet/coreclr#26025]
```
GC/API/GC/KeepAliveNull/KeepAliveNull.sh [FAIL]
      
      Assert failure(PID 24052 [0x00005df4], Thread: 24052 [0x5df4]): currentProcessCpuCount == g_processAffinitySet.Count()
          File: /__w/2/s/src/vm/gcenv.os.cpp Line: 114
          Image: /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload/corerun
      
      /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/GC/KeepAliveNull/KeepAliveNull.sh: line 252: 24052 Aborted                 $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"
      
      Return code:      1
      Raw output file:      /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/Reports/GC.API/GC/KeepAliveNull/KeepAliveNull.output.txt
      Raw output:
      BEGIN EXECUTION
      /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload/corerun KeepAliveNull.exe ''
      Expected: 100
      Actual: 134
      END EXECUTION - FAILED
      Test Harness Exitcode is : 1
      To run the test:
      > set CORE_ROOT=/ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload
      > /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/GC/KeepAliveNull/KeepAliveNull.sh
      Expected: True
      Actual:   False
      Stack Trace:
        /__w/2/s/bin/tests/Linux.arm.Checked/TestWrappers/GC.API/GC.API.XUnitWrapper.cs(2856,0): at GC_API._GC_KeepAliveNull_KeepAliveNull_._GC_KeepAliveNull_KeepAliveNull_sh()
      Output:
        
        Assert failure(PID 24052 [0x00005df4], Thread: 24052 [0x5df4]): currentProcessCpuCount == g_processAffinitySet.Count()
            File: /__w/2/s/src/vm/gcenv.os.cpp Line: 114
            Image: /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload/corerun
        
        /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/GC/KeepAliveNull/KeepAliveNull.sh: line 252: 24052 Aborted                 $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"
        
        Return code:      1
        Raw output file:      /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/Reports/GC.API/GC/KeepAliveNull/KeepAliveNull.output.txt
        Raw output:
        BEGIN EXECUTION
        /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload/corerun KeepAliveNull.exe ''
        Expected: 100
        Actual: 134
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        To run the test:
        > set CORE_ROOT=/ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Payload
        > /ssd/dotnetbuild/work/66290456-48d2-448c-ae61-9d5ab09b9d72/Work/ea6de819-2a26-464b-aba9-057e7137059f/Exec/GC/API/GC/KeepAliveNull/KeepAliveNull.sh
```

**Comment by jashook** [comment]
https://helix.dot.net/api/2019-06-17/jobs/66290456-48d2-448c-ae61-9d5ab09b9d72/workitems/PayloadGroup0/console

**Comment by jashook** [comment]
https://dev.azure.com/dnceng/public/_build/results?buildId=296117

**Comment by Maoni0** [comment]
@janvorli would you please take a  look?

**Comment by jashook** [comment]
Thank you for routing :)

**Comment by janvorli** [comment]
This issue is a dup of #24050
This is a known problem happening on our lab NVidia Jetson TX1 devices. These devices are configured to dynamically bring cores on and offline based on the system load. This gets dynamically reflected in the affinity set of the current process. When the load changes between the call to PAL_GetLogicalCpuCountFromOS and PAL_GetCurrentThreadAffinitySet, the assert fires as the affinity mask contains a different number of bits than the number we got from the PAL_GetLogicalCpuCountFromOS. Even worse, the PAL_GetLogicalCpuCountFromOS basically gets a random number from 1 to the real number of cores based on the system load at .NET Core app startup.
On some devices with the same behavior, this could be mitigated by configuring the cpu scaling governor to "performance" mode which keeps all cores online. 
I've asked the eng team to configure our lab TX1 devices to use that, but I've learned that we are in a process of moving away from using the TX1 and to using TX2 devices instead. The TX2 doesn't have this problem, so we've decided to do nothing for now.
